src/handler/SpecifyBootVersionStep.ts

```diff
@@ -29,13 +29,21 @@ export class SpecifyBootVersionStep implements IStep {
     }
 
     private async specifyBootVersion(projectMetadata: IProjectMetadata): Promise<boolean> {
+        const items = await serviceManager.getItems(projectMetadata.serviceUrl, MetadataType.BOOTVERSION);
+
+        if (projectMetadata.enableSmartDefaults === true) {
+            projectMetadata.bootVersion = items.find(x => x.default === true)?.value?.id;
```

Comment:
we should guard the case of `projectMetadata.bootVersion` is null, and fall back to the prompt mode.

Reply:
Fixed that for every API default. Thanks!